### PR TITLE
Guard check with deployment type

### PR DESCRIPTION
**Pull request by ubreu**
Only apply check if deployment type is 'atomic-openshift'

### CODE REVIEW

File: playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@
 
   - fail:
       msg: Atomic OpenShift 3.1 packages not found
-    when: g_aos_versions.curr_version | version_compare('3.0.2.900','<') and (g_aos_versions.avail_version is none or g_aos_versions.avail_version | version_compare('3.0.2.900','<'))
+    when: deployment_type == 'atomic-openshift' and (g_aos_versions.curr_version | version_compare('3.0.2.900','<') and (g_aos_versions.avail_version is none or g_aos_versions.avail_version | version_compare('3.0.2.900','<')))
```
Comment (abutcher) [inline playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml]: I think this should check for `openshift-enterprise` deployment type as well.

`deployment_type in ['atomic-openshift', 'openshift-enterprise']`

Comment (detiber) [inline playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml]: We don't yet support upgrades for atomic-enterprise, so the check should be for deployment_type in ['online', 'openshift-enterprise']

There would also need to be an additional check to make sure the origin package is gte 1.1

**Comment by ubreu** [comment]
@detiber : Do you mean that the origin check should be amended as follows:
```
  - fail:
      msg: This playbook requires Origin 1.0.6 or later and the Origin 1.1 package must be available
    when: deployment_type == 'origin' and (g_aos_versions.curr_version | version_compare('1.0.6','<') or (g_aos_versions.avail_version is none or g_aos_versions.avail_version | version_compare('1.1','<')))
```

Regarding the deployment_type guards: There already is a guard on lines 37 - 41:
```
  - fail:
      msg: >
        This upgrade is only supported for origin and openshift-enterprise
        deployment types
    when: deployment_type not in ['origin','openshift-enterprise']
```
So should the guard actually be:
```
- fail:
      msg: Atomic OpenShift 3.1 packages not found
    when: deployment_type == 'openshift-enterprise' and (g_aos_versions.curr_version | version_compare('3.0.2.900','<') and (g_aos_versions.avail_version is none or g_aos_versions.avail_version | version_compare('3.0.2.900','<')))
```

**Comment by detiber** [comment]
@ubreu that looks correct for both

**Comment by brenton** [comment]
@detiber, is there a cleaner way to structure these `when` conditionals?  Perhaps putting them on multiple lines and adding comments?  There are some subtle differences with the check between Origin and OSE involving how we enforce the minimum version required and I'm worried that future modifications to this logic will break something unless the intentions are more clear.

**Comment by detiber** [comment]
@brenton I'm not sure how the when condition handles being spread across multiple lines, since it is an implicit jinja2 template. Instead we should probably replace the complex conditions with a filter. I'll see about working something up.

**Comment by detiber** [comment]
@brenton something like this should work (warning it is untested): https://github.com/openshift/openshift-ansible/pull/1008 I was able to rework it a bit without resorting to a filter plugin.

The downside is the loss of treating 3.0.2.900 as an upgrade, but since that isn't a supported version to upgrade to I think the simplification makes sense.

**Comment by ubreu** [comment]
@brenton: I guess we can close this pull request?

**Pull request closed without merging**